webdictionary/src/main/java/com/eis0/kademlianetwork/informationdeliverymanager/IdFinderHandler.java

```diff
@@ -0,0 +1,184 @@
+     * continue the research; if the closerNode find the searched ID, it sends it directly to the
+     * node which originated the research, without retracing back all the node that took part to the
+     * research
+     *
+     * @param idToFind     The ID whose research originated the request
+     *                     It's sent back as response to the research
+     * @param searcherNode The Peer which started the research
+     * @param closerNode   The node with the {@link KademliaId} closer to the idToFind
+     */
+    private static void keepLooking(KademliaId idToFind, SMSPeer searcherNode, SMSPeer closerNode) {
+        SMSMessage requestMessage = new KademliaMessage()
+                .setPeer(closerNode)
+                .setRequestType(RequestTypes.FindId)
+                .setIdToFind(idToFind)
+                .setSearcher(searcherNode)
+                .buildMessage();
+        SMSManager.getInstance().sendMessage(requestMessage);
+    }
+
+    /**
+     * This method verify if the nodeToCheck is still valid. If the node returns any sign of life
+     * the process ends, otherwise the research for the ID restarted
+     *
+     * @param idToFind     The ID to find inside the network
+     * @param searcherNode The Peer which is searching the ID
+     * @param nodeToCheck  The Peer whose validity must be checked; if it's not active
+     *                     anymore, it is removed from the RoutingTable, and the research is started
+     *                     again
+     */
+    private static void retryIfDead(KademliaId idToFind, SMSPeer searcherNode, SMSPeer nodeToCheck,
+                                    List<SMSKademliaNode> nodeList) {
+        if (!KademliaJoinableNetwork.getInstance().isAlive(nodeToCheck)) {
+            //The first node is the one tested, if I'm here it failed.
+            //I remove it and restart the process
+            nodeList.remove(0);
+            //Recursive call
+            searchIdList(idToFind, searcherNode, nodeList);
+        }
+    }
+
+    /**
+     * The number of maximum attempts for the research of the node in the RoutingTable is by default
+     * set to 10; the reason for the existence of this value is to assure that, if the node can't
+     * reach a large number of nodes, it can still insert the resource inside of the Network; the
+     * resource will be then relocated during the refresh, IF the unresponsive node returns to live.
+     * If it doesn't, then the node chosen as closest one remains the closest one.
+     * It must also be considered the time that each attempts takes (10s)
+     * Warning: when the application is closed the number of attempts is reset
+     *
+     * @param attemptsNumber The maximum number of attempts before saving the resource in the local
+     *                       dictionary
+     */
+    private static void setAttemptsNumber(int attemptsNumber) {
```

Comment:
Why have a method to change the number of attempts if it is private?

Reply:
Oh
My fault